pyccel/parser/semantic.py

```diff
@@ -3185,7 +3190,8 @@ def _visit_Return(self, expr):
 
         # add the Deallocate node before the Return node and eliminating the Deallocate nodes
         # the arrays that will be returned.
-        code = assigns + [Deallocate(i) for i in self._allocs[-1] if i not in results]
+        code = assigns + [Deallocate(i if not isinstance(i, ConstructorCall) else i.cls_variable) for i in self._allocs[-1] \
+                            if (i if not isinstance(i, ConstructorCall) else i.cls_variable) not in results]
```

Comment:
Is this change still necessary?

Reply:
Fixed 0dbb55d7043fb5c0772d201935163c592e7097eb.